lib/catalog/order_item_sanitized_parameters.rb

```diff
@@ -69,22 +64,52 @@ def mask_value?(field)
     end
 
     def fields
-      service_plan_schema.dig(:schema, :fields) || []
+      @fields ||= service_plan_schema.with_indifferent_access.dig(:schema, :fields)
     end
 
     def service_plan_does_not_exist?
       service_plan_ref.nil?
     end
 
+    def sanitize_value(key, value)
+      field = fields.find { |f| f[:name] == key }
+      return value unless field[:isSubstitution]
```

Comment:
@bzwei Should we check if we got a field back this might work from UI but if someone send it directly via a REST API, the field might not exist. Also the base schema might not have isSubstitution.

Reply:
Without this flag we do not do substitution. If modified does not exist, we purely rely one base and no substitution will happen. For a REST API user he need to create a modified schema in the first place.

Reply:
@bzwei If fields.find comes back nil that field[:isSubstitution] is going to blow up

Reply:
the field is guaranteed by https://github.com/RedHatInsights/catalog-api/pull/843/files#diff-32b667336a8a0f30ae4e30c6b14d976fdaa183eb28bf254c3c9eca85c140ba5bR40